parser-typechecker/src/Unison/Codebase/Editor/Output.hs

```diff
@@ -81,7 +82,7 @@ data Output v
   | TypeHasFreeVars Input (Type v Ann)
   | TermAlreadyExists Input Path.Split' (Set Referent)
   | TypeAmbiguous Input Path.HQSplit' (Set Reference)
-  | TermAmbiguous Input Path.HQSplit' (Set Referent)
+  | TermAmbiguous Input (Either Path.HQSplit' HQ.HashQualified) (Set Referent)
```

Comment:
Clearly we need to refactor these types.

Reply:
`Path.HQSplit'` and `HQ.HashQualified` can overlap.  Did you mean `Either Path.HQSplit' ShortHash`?

Reply:
I dug into this a bit more.  First, the data in `TermAmbiguous` doesn't ever get displayed, so it's all kinda moot.  Second, the `Either` here is to overload `TermAmbiguous` for a different case when the ambiguous thing you have is a `HashQualified` @runarorama